foundations/foundation-metrics/src/main/java/io/servicecomb/foundation/metrics/output/servo/ServoMetricsFileOutput.java

```
@@ -0,0 +1,154 @@
+  public void init() {
+    PollScheduler scheduler = PollScheduler.getInstance();
+    if (!scheduler.isStarted()) {
+      scheduler.start();
+    }
+
+    if (isEnabled()) {
+      MetricObserver fileObserver = new SeparatedMetricObserver(applicationName, this, registry);
+      MetricObserver fileTransform = new CounterToRateMetricTransform(fileObserver, getMetricPoll(), TimeUnit.SECONDS);
+      PollRunnable fileTask = new PollRunnable(new MonitorRegistryMetricPoller(), BasicMetricFilter.MATCH_ALL,
+          fileTransform);
+      scheduler.addPoller(fileTask, getMetricPoll(), TimeUnit.SECONDS);
+    }
+  }
+
+  @Override
+  public void output(Map<String, String> metrics) {
+
+    for (String metricName : metrics.keySet()) {
+      final String fileName = String.join(".", this.applicationName, metricName, "dat");
+      RollingFileAppenderExt appender = metricsAppenders.computeIfAbsent(metricName, (key) -> {
+        String finalPath = Paths.get(getRollingRootFilePath(), fileName).toString();
+        RollingFileAppenderExt fileAppender = new RollingFileAppenderExt();
+        fileAppender.setLogPermission("rw-------");
+        fileAppender.setFile(finalPath);
+        fileAppender.setLayout(new PatternLayout("%m%n"));
+        fileAppender.setThreshold(Priority.FATAL);
+        fileAppender.setAppend(true);
+        fileAppender.setMaxFileSize(getMaxRollingFileSize());
+        fileAppender.setMaxBackupIndex(getMaxRollingFileCount());
+        fileAppender.activateOptions();
+        return fileAppender;
+      });
+
+      LoggingEvent event = null;
+      try {
+        SeparatedOutputData outputData = new SeparatedOutputData(this.applicationName, hostName, metricName,
+            metrics.get(metricName));
+        event = new LoggingEvent(fileName, Category.getInstance(fileName), Priority.FATAL,
+            mapper.writeValueAsString(outputData), null);
+      } catch (JsonProcessingException e) {
+        logger.error("parse metric data error");
```

Do not eating the exception!!!

---

Fixed